src/lib.rs

```diff
@@ -242,7 +242,7 @@ impl<'a, R: Rng> MarkovChain<'a, R> {
             map: &self.map,
             rng: &mut self.rng,
             keys: &self.keys,
-            state: state,
```

Comment:
Please put this back — I like the explicit field names since it matches the other lines.